apps/settings/lib/SetupChecks/TaskProcessingPickupSpeed.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace OCA\Settings\SetupChecks;
+
+use OCP\AppFramework\Utility\ITimeFactory;
+use OCP\IL10N;
+use OCP\SetupCheck\ISetupCheck;
+use OCP\SetupCheck\SetupResult;
+use OCP\TaskProcessing\IManager;
+
+class TaskProcessingPickupSpeed implements ISetupCheck {
+	public function __construct(
+		private IL10N $l10n,
+		private IManager $taskProcessingManager,
+		private ITimeFactory $timeFactory,
+	) {
+	}
+
+	public function getCategory(): string {
+		return 'ai';
+	}
+
+	public function getName(): string {
+		return $this->l10n->t('Task Processing pickup speed');
+	}
+
+	public function run(): SetupResult {
+		$tasks = $this->taskProcessingManager->getTasks(userId: '', scheduleAfter: $this->timeFactory->now()->getTimestamp() - 60 * 60 * 24); // userId: '' means no filter, whereas null would mean guest
```

Comment:
```suggestion
		// this could also be a class constant
		$nbHoursToCheck = 24;
		$tasks = $this->taskProcessingManager->getTasks(userId: '', scheduleAfter: $this->timeFactory->now()->getTimestamp() - 60 * 60 * $nbHoursToCheck); // userId: '' means no filter, whereas null would mean guest
```
and use it everywhere.